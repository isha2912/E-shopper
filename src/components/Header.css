.nav-bar {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  background-color: black;
  height: 100%;
  width: 100%;
  align-items: center;
  color: white;
}
.nav-bar > div {
  padding: 1rem;
}
.logo {
  flex-grow: 3;
  color: white;
}
.logo > img {
  height: 4rem;
  width: 5rem;
  margin-left: 1rem;
}

.nav-right {
  display: flex;
  justify-content: space-between;
}
.all-product-route {
  color: white;
}
.cart-count-route {
  color: white;
}
.cart-count {
  margin-left: 1rem;
  display: flex;
  justify-content: space-between;
}
