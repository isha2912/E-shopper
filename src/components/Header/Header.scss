.nav-bar {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  background-color: black;
  height: 100%;
  width: 100%;
  align-items: center;
  color: white;
}
.nav-bar-light {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  background-color: white;
  height: 100%;
  width: 100%;
  align-items: center;
  color: black;
}
.nav-bar > div {
  padding: 1rem;
}
.nav-bar-light > div {
  padding: 1rem;
}
.logo {
  flex-grow: 3;
  color: white;
}
.logo > img {
  height: 4rem;
  width: 5rem;
  margin-left: 1rem;
}

.nav-right {
  display: flex;
  justify-content: space-between;
}
.all-product-route {
  color: white;
}
.all-product-route-white {
  color: black;
}
.cart-count-route {
  color: white;
}
.cart-count-route-white {
  color: black;
}
.cart-count {
  margin-left: 1rem;
  display: flex;
  justify-content: space-between;
}
