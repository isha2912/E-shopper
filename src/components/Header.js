/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { useContext } from 'react';
import './Header.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { ThemeContext } from './ThemeContext';

const Header = (props) => {
  const { cartCount } = props;
  const theme = useContext(ThemeContext);
  return (

    <nav className={theme ? 'nav-bar' : 'nav-bar-light'}>
      <Link to="/">
        {' '}
        <div className="logo">
          <img data-testid="img-tag" src="./assets/download (1).jpeg" alt="logo" />
        </div>
      </Link>
      <div className="nav-right">
        <Link to="/allorders">
          {' '}
          <h5 className={theme ? 'all-product-route' : 'all-product-route-white'}> ALL PRODUCTS </h5>
          {' '}
        </Link>
        <div>
          <div className="cart-count">
            <Link to="/cart">
              {' '}
              <h5 className={theme ? 'cart-count-route' : 'cart-count-route-white'}> CART COUNT:</h5>
              {' '}
            </Link>
            <h5 className={theme ? 'cart-count-route' : 'cart-count-route-white'}>
              {' '}
              {cartCount}
              {' '}
            </h5>
          </div>
          {' '}
        </div>
      </div>
    </nav>
  );
};

Header.proptype = {
  cartCount: PropTypes.number.isRequired,
};
export default Header;
