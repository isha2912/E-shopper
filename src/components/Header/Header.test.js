/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

describe('Should display the nav bar', () => {
  const mockProp = {
    cartCount: 0,
  };
  afterEach(() => jest.clearAllMocks());

  it('Should display logo, cart count and All products', () => {
    const { container } = render(
      <BrowserRouter>
        <Header {...mockProp} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('Should take to Home page on click', () => {
    render(
      <BrowserRouter>
        <Header {...mockProp} />
      </BrowserRouter>,
    );
    const logoLink = screen.getByTestId('img-tag');
    fireEvent.click(logoLink);
    expect(document.location.href).toBe('http://localhost/');
  });
  it('Should take to Cart page on click', () => {
    render(
      <BrowserRouter>
        <Header {...mockProp} />
      </BrowserRouter>,
    );
    const cartLink = screen.getByText('CART COUNT:');
    fireEvent.click(cartLink);
    expect(document.location.href).toBe('http://localhost/cart');
  });
  it('Should take to All product page on click', () => {
    render(
      <BrowserRouter>
        <Header {...mockProp} />
      </BrowserRouter>,
    );
    const allProductLink = screen.getByText('ALL PRODUCTS');
    fireEvent.click(allProductLink);
    expect(document.location.href).toBe('http://localhost/allorders');
  });
});
