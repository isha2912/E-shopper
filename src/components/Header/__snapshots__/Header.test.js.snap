// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the nav bar Should display logo, cart count and All products 1`] = `
<div>
  <nav
    class="nav-bar"
  >
    <a
      href="/"
    >
       
      <div
        class="logo"
      >
        <img
          alt="logo"
          data-testid="img-tag"
          src="./assets/download (1).jpeg"
        />
      </div>
    </a>
    <div
      class="nav-right"
    >
      <a
        href="/allorders"
      >
         
        <h5
          class="all-product-route"
        >
           ALL PRODUCTS 
        </h5>
         
      </a>
      <div>
        <div
          class="cart-count"
        >
          <a
            href="/cart"
          >
             
            <h5
              class="cart-count-route"
            >
               CART COUNT:
            </h5>
             
          </a>
          <h5
            class="cart-count-route"
          >
             
            0
             
          </h5>
        </div>
         
      </div>
    </div>
  </nav>
</div>
`;
